src/lib/builtins/array.rs

```diff
@@ -500,10 +512,9 @@ pub fn fill(this: &Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
         cmp::min(relative_end, len)
     };
 
-    while k < fin {
-        let pk = k.to_string();
+    for i in k..fin {
+        let pk = i.to_string();
         this.set_field_slice(&pk, value.clone());
```

Comment:
This should be `set_field`. Using `set_field_slice` does an extra allocation here.